src/Decompiler/Analysis/UserSignatureBuilder.cs

```diff
@@ -160,5 +161,52 @@ public ProcedureBase_v1 ParseFunctionDeclaration(string fnDecl, Frame frame)
                 return null;
             }
         }
+
+        public GlobalDataItem_v2 ParseGlobalDeclaration(string txtGlobal)
+        {
+            try
+            {
+                var tokens = txtGlobal.Split(new char[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
+                if (tokens.Length <= 1)
+                    return null;
+                var name = tokens[tokens.Length - 1];
+                if (!IsValidCIdentifier(name))
+                    return null;
+                var lexer = new CLexer(new StringReader("typedef " + txtGlobal + ";"));
+                var symbols = program.CreateSymbolTable();
+                var oldNamedTypes = symbols.NamedTypes.Count;
+                var cstate = new ParserState(symbols.NamedTypes.Keys);
+                var cParser = new CParser(cstate, lexer);
+                var decl = cParser.Parse_ExternalDecl();
+                if (decl == null)
+                    return null;
+
+                //$HACK: Relying on a side effect here to
+                // get both the global type. Ew.
+                symbols.AddDeclaration(decl);
+                if (symbols.NamedTypes.Count == oldNamedTypes)
+                    return null;
+                var type = symbols.NamedTypes[name];
+                var global = new GlobalDataItem_v2()
+                {
+                    Name = name,
+                    DataType = type,
+                };
+                return global;
+            }
+            catch (Exception ex)
+            {
+                //$TODO: if user has supplied a global variable that can't parse,
+                // we must notify them in the diagnostics window with a 
+                // WARNING.
```

Comment:
What I mean is that since you are changing the color as the user is typing, there is no need to litter the diagnostics with lots of warnings... Unless you generate a warning, like
```'foo@dbl' is not a valid C identifier```
and as the user types / corrects, you update the warning
```'foo@' is not a valid C identifier```
until the final @ is removed and the error is removed from the diagnostics list.